### PR TITLE
Fix for Broken optimization path on indexed fixtures

### PR DESCRIPTION
**Pull request by ctsims**
The `[selected(list, indexed_reference)]` predicate is coded to be optimizable for case index lookups. That predicate's existence is breaking the optimization route when it doesn't get used due to a bug in processing.

The attached test replicates the bug.

Useful to fix in 2.45, because REACH likely will be using this reference pattern.

### CODE REVIEW

File: src/main/java/org/commcare/cases/util/StorageBackedTreeRoot.java [hunk 1]
```diff
@@ -286,7 +286,7 @@ protected void initBasicQueryHandlers(QueryPlanner queryPlanner) {
                                                     QueryContext currentQueryContext) throws IllegalArgumentException {
         int numKeysToProcess = this.getNumberOfBatchableKeysInProfileSet(profiles);
 
-        if(numKeysToProcess == -1) {
+        if(numKeysToProcess  < 1) {
```
Comment (ctsims) [inline src/main/java/org/commcare/cases/util/StorageBackedTreeRoot.java]: core fix

File: src/main/java/org/javarosa/core/model/instance/utils/FormLoadingUtils.java [hunk 2]
```diff
@@ -27,13 +27,20 @@ public static FormInstance loadFormInstance(String formFilepath)
 
     public static TreeElement xmlToTreeElement(String xmlFilepath)
             throws InvalidStructureException, IOException {
-        InputStream is = FormLoadingUtils.class.getResourceAsStream(xmlFilepath);
-        TreeElementParser parser = new TreeElementParser(ElementParser.instantiateParser(is), 0, "instance");
-
+        InputStream is = null;
```
Comment (ctsims) [inline src/main/java/org/javarosa/core/model/instance/utils/FormLoadingUtils.java]: This change just fixes the test runner

File: src/test/java/org/commcare/fixtures/test/IndexedFixtureTests.java [hunk 3]
```diff
@@ -109,5 +109,13 @@ private void doPartialLookupTest() throws XPathSyntaxException, UnfullfilledRequ
...
     }
 
+    @Test
+    public void queryInSetLookup() throws XPathSyntaxException, UnfullfilledRequirementsException,
+            XmlPullParserException, IOException, InvalidStructureException {
+        ParseUtils.parseIntoSandbox(getClass().getResourceAsStream("/indexed-fixture-with-keys.xml"), sandbox);
 
+        EvaluationContext ec =
+                MockDataUtils.buildContextWithInstance(sandbox, "products", CaseTestUtils.FIXTURE_INSTANCE_PRODUCT);
+        CaseTestUtils.xpathEvalAndAssert(ec, "sort(join(' ', instance('products')/products/product[selected('a6d16035b98f6f962a6538bd927cefb3 31ab899368d38c2d0207fe80c00fb8c1', @id)]/name))", "CU DIU");
```
Comment (ctsims) [inline src/test/java/org/commcare/fixtures/test/IndexedFixtureTests.java]: bug replication

**Comment by shubham1g5** [comment]
@ctsims  The build Failure looks real. From Jenkins - 
```
org.commcare.fixtures.test.IndexedFixtureTests > queryInSetLookup FAILED
 java.io.IOException at IndexedFixtureTests.java:115
```

**Comment by ctsims** [comment]
@shubham1g5 Fixed, thanks for catching.

**Pull request merged**